standalone-metastore/metastore-catalog/src/main/java/org/apache/iceberg/rest/HMSCatalogServer.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest;
+
+import java.io.IOException;
+import java.lang.ref.Reference;
+import java.lang.ref.SoftReference;
+import java.util.Collections;
+import java.util.Map;
+import java.util.TreeMap;
+import javax.servlet.http.HttpServlet;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hive.metastore.SecureServletCaller;
+import org.apache.hadoop.hive.metastore.ServletSecurity;
+import org.apache.hadoop.hive.metastore.conf.MetastoreConf;
+import org.apache.iceberg.HiveCachingCatalog;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.hive.HiveCatalog;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.handler.gzip.GzipHandler;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class HMSCatalogServer {
+  private static final String CACHE_EXPIRY = "hive.metastore.catalog.cache.expiry";
+  /**
+   * The metric names prefix.
+   */
+  static final String HMS_METRIC_PREFIX = "hmscatalog.";
+  private static final Logger LOG = LoggerFactory.getLogger(HMSCatalogServer.class);
+  private static Reference<Catalog> catalogRef;
+
+  static Catalog getLastCatalog() {
+    return catalogRef != null ? catalogRef.get() :  null;
+  }
+
+  private HMSCatalogServer() {
+    // nothing
+  }
+
+  public static HttpServlet createServlet(SecureServletCaller security, Catalog catalog) throws IOException {
+    try (HMSCatalogAdapter adapter = new HMSCatalogAdapter(catalog)) {
```

Comment:
nit: why do we need the try-with-resource, it sounds like the `adapter` is invalid after we create the `HMSCatalogServlet`.

Reply:
Corrected.